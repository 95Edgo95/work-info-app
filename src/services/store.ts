import createSagaMiddleware, {SagaMiddleware} from "redux-saga";
import {applyMiddleware, compose, createStore} from "redux";
import reducer from "services/reducer";
import saga from "services/saga";
import {fromJS} from "immutable";

let store: any;

const initialState: any = fromJS({});

const sagaMiddleware: SagaMiddleware<any> = createSagaMiddleware();

if (process.env && process.env.NODE_ENV !== "production") {
    const composeEnhancers: any = (window as any).__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;
    store = createStore(reducer, initialState, composeEnhancers(applyMiddleware(sagaMiddleware)));
} else {
    store = createStore(reducer, initialState, compose(applyMiddleware(sagaMiddleware)));
}

sagaMiddleware.run(saga);

export default store;
