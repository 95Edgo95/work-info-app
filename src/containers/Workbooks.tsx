import {attemptDeleteWorkbook, attemptGetWorkbooks} from "store/workbook/WorkbookActionCreators";
import {resetWorkplaceData} from "store/workplace/WorkplaceActionCreators";
import {mapDispatchToProps, mapStateToProps} from "helpers/ReduxHelper";
import {IWorkbook} from "store/workbook/WorkbookReducer";
import {RouteComponentProps} from "react-router-dom";
import {ISelectorVariables} from "services/selector";
import EntityActions from "components/EntityActions";
import CardsWrapper from "containers/CardsWrapper";
import Filter from "components/Filter";
import {connect} from "react-redux";
import * as React from "react";
import {Map} from "immutable";

interface IWorkbooksReduxProps {
  workbooks: Map<string, IWorkbook>;
  isLoading: boolean;
  pagination: any;
  role: string;
}

interface IWorkbooksDispatchProps {
  attemptDeleteWorkbook: (id: string) => void;
  attemptGetWorkbooks: (params: any) => void;
  resetWorkplaceData: () => void;
}

interface IWorkbooksProps extends IWorkbooksReduxProps, IWorkbooksDispatchProps, RouteComponentProps {}

function Workbooks(props: IWorkbooksProps) {
  const {
    attemptDeleteWorkbook,
    attemptGetWorkbooks,
    resetWorkplaceData,
    pagination,
    workbooks,
    isLoading,
    history,
    role,
  } = props;

  React.useEffect(() => {
    resetWorkplaceData();
  }, []);

  return (
    <React.Fragment>
      <Filter
        attemptGetEntities={attemptGetWorkbooks}
        showAddNew={role === "admin"}
        createLink="workbooks"
        inputs={[
          {
            placeholder: "First Name",
            name: "firstName",
            defaultValue: "",
          },
          {
            placeholder: "Last Name",
            defaultValue: "",
            name: "lastName",
          },
          {
            placeholder: "Passport",
            defaultValue: "",
            name: "passport",
          },
          {
            placeholder: "Email",
            defaultValue: "",
            name: "email",
          }
        ]}
      />
      <CardsWrapper
        attemptGetEntities={attemptGetWorkbooks}
        entities={workbooks}
        getDetails={workbook => [
          {
            value: workbook.get("email"),
            tooltip: "Email",
            icon: "email",
          },
          {
            value: workbook.get("passport"),
            tooltip: "Passport",
            icon: "passport",
          },
          {
            value: workbook.get("birthDate"),
            tooltip: "Birth Date",
            icon: "birthday",
          },
          {
            value: workbook.get("workplacesIds").size,
            tooltip: "Workplaces count",
            icon: "workplace",
          },
        ]}
        getTitle={workbook => `${workbook.get("firstName")} ${workbook.get("lastName")}`}
        getLink={workbook => `/workplaces/${workbook.get("id")}`}
        showActions={role === "admin"}
        pagination={pagination}
        isLoading={isLoading}
      >
        {id => (
          <EntityActions
            handleDelete={(event) => {
              event.preventDefault();
              attemptDeleteWorkbook(id)
            }}
            handleEdit={(event) => {
              event.preventDefault();
              history.push(`/workbooks/${id}/edit`);
            }}
          />
        )}
      </CardsWrapper>
    </React.Fragment>
  );
}

const selectorVariables: ISelectorVariables = {
  workbook: {
    workbooks: true,
    pagination: true,
    isLoading: true
  },
  user: {
    role: true,
  },
};

const actions = {
  attemptDeleteWorkbook,
  attemptGetWorkbooks,
  resetWorkplaceData,
};

export default connect<IWorkbooksReduxProps, IWorkbooksDispatchProps, any>(mapStateToProps(selectorVariables), mapDispatchToProps(actions))(Workbooks)
