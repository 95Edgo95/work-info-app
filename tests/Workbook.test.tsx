import Workbooks from "../src/containers/Workbooks";
import {BrowserRouter} from "react-router-dom";
import store from "../__mocks__/storeMock";
import {Provider} from "react-redux";
import { mount } from "enzyme";
import * as React from "react";

describe('Workbook', () => {
  it('should render correctly', () => {
    const component = mount(
      <Provider store={store}>
        <BrowserRouter>
          <Workbooks />
        </BrowserRouter>
      </Provider>
    );

    expect(component).toMatchSnapshot();
    component.unmount();
  });
});
