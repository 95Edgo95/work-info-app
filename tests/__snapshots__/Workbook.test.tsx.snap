// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Workbook should render correctly 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <BrowserRouter>
    <Router
      history={
        Object {
          "action": "POP",
          "block": [Function],
          "createHref": [Function],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        }
      }
    >
      <Connect(Workbooks)>
        <Workbooks
          attemptDeleteWorkbook={[Function]}
          attemptGetWorkbooks={[Function]}
          pagination={
            Immutable.Map {
              "count": 0,
              "limit": 0,
              "offset": 0,
            }
          }
          resetWorkplaceData={[Function]}
          workbooks={Immutable.Map {}}
        >
          <Filter
            attemptGetEntities={[Function]}
            createLink="workbooks"
            inputs={
              Array [
                Object {
                  "defaultValue": "",
                  "name": "firstName",
                  "placeholder": "First Name",
                },
                Object {
                  "defaultValue": "",
                  "name": "lastName",
                  "placeholder": "Last Name",
                },
                Object {
                  "defaultValue": "",
                  "name": "passport",
                  "placeholder": "Passport",
                },
                Object {
                  "defaultValue": "",
                  "name": "email",
                  "placeholder": "Email",
                },
              ]
            }
          >
            <div>
              <Input
                key="firstName"
                name="firstName"
                onChange={[Function]}
                placeholder="First Name"
                value=""
              >
                <div>
                  <div
                    className=""
                  >
                    <input
                      name="firstName"
                      onChange={[Function]}
                      placeholder="First Name"
                      type="text"
                      value=""
                    />
                  </div>
                </div>
              </Input>
              <Input
                key="lastName"
                name="lastName"
                onChange={[Function]}
                placeholder="Last Name"
                value=""
              >
                <div>
                  <div
                    className=""
                  >
                    <input
                      name="lastName"
                      onChange={[Function]}
                      placeholder="Last Name"
                      type="text"
                      value=""
                    />
                  </div>
                </div>
              </Input>
              <Input
                key="passport"
                name="passport"
                onChange={[Function]}
                placeholder="Passport"
                value=""
              >
                <div>
                  <div
                    className=""
                  >
                    <input
                      name="passport"
                      onChange={[Function]}
                      placeholder="Passport"
                      type="text"
                      value=""
                    />
                  </div>
                </div>
              </Input>
              <Input
                key="email"
                name="email"
                onChange={[Function]}
                placeholder="Email"
                value=""
              >
                <div>
                  <div
                    className=""
                  >
                    <input
                      name="email"
                      onChange={[Function]}
                      placeholder="Email"
                      type="text"
                      value=""
                    />
                  </div>
                </div>
              </Input>
              <Button
                onClick={[Function]}
              >
                <button
                  className=""
                  onClick={[Function]}
                  type="button"
                >
                  Search
                </button>
              </Button>
              <Button
                secondary={true}
              >
                <button
                  className=""
                  type="button"
                >
                  <Link
                    replace={false}
                    to="/workbooks/create"
                  >
                    <a
                      href="/workbooks/create"
                      onClick={[Function]}
                    >
                      Add New
                    </a>
                  </Link>
                </button>
              </Button>
            </div>
          </Filter>
          <CardsWrapper
            attemptGetEntities={[Function]}
            entities={Immutable.Map {}}
            getDetails={[Function]}
            getLink={[Function]}
            getTitle={[Function]}
            pagination={
              Immutable.Map {
                "count": 0,
                "limit": 0,
                "offset": 0,
              }
            }
          >
            <div>
              <Component />
            </div>
          </CardsWrapper>
        </Workbooks>
      </Connect(Workbooks)>
    </Router>
  </BrowserRouter>
</Provider>
`;
